parts/linux/cloud-init/artifacts/cse_config.sh

```diff
@@ -384,6 +386,21 @@ configureContainerdRegistryHost() {
 EOF
 }
 
+configureContainerdLegacyMooncakeMcrHost() {
+    LEGACY_MCR_REPOSITORY_BASE="mcr.azk8s.cn"
+    CONTAINERD_CONFIG_REGISTRY_HOST_MCR="/etc/containerd/certs.d/${LEGACY_MCR_REPOSITORY_BASE}/hosts.toml"
+    mkdir -p "$(dirname "${CONTAINERD_CONFIG_REGISTRY_HOST_MCR}")"
+    touch "${CONTAINERD_CONFIG_REGISTRY_HOST_MCR}"
+    chmod 0644 "${CONTAINERD_CONFIG_REGISTRY_HOST_MCR}"
+
+    TARGET_MCR_REPOSITORY_BASE="mcr.azure.cn"
+    tee "${CONTAINERD_CONFIG_REGISTRY_HOST_MCR}" > /dev/null <<EOF
+[host."https://${TARGET_MCR_REPOSITORY_BASE%/}"]
+  capabilities = ["pull", "resolve"]
+  override_path = true
```

Comment:
Not needed for OCI-complaint regitries

https://github.com/containerd/containerd/blob/main/docs/hosts.md#override_path-field
```suggestion
```